execution-plane/arcon/src/stream/source/collection.rs

```diff
@@ -52,21 +72,35 @@ where
 {
     fn handle(&mut self, event: ControlEvent) {
         if let ControlEvent::Start = event {
-            self.process_collection();
+            let shared = self.loopback_receive.share();
+            self.loopback_send.connect(shared);
+            self.loopback_send.trigger(ContinueSending);
         }
     }
 }
 
-impl<IN, OUT> Actor for CollectionSource<IN, OUT>
+impl<IN, OUT> Provide<LoopbackPort> for CollectionSource<IN, OUT>
 where
     IN: ArconType,
     OUT: ArconType,
 {
-    type Message = ();
-    fn receive_local(&mut self, _msg: Self::Message) {}
-    fn receive_network(&mut self, _msg: NetMessage) {}
+    fn handle(&mut self, _event: ContinueSending) {
+        self.process_collection();
+    }
 }
 
+impl<IN, OUT> Require<LoopbackPort> for CollectionSource<IN, OUT>
+where
+    IN: ArconType,
+    OUT: ArconType,
+{
+    fn handle(&mut self, _event: Never) {
+        unreachable!("Never type has no instance");
+    }
+}
+
+//ignore_indications!(LoopbackPort, CollectionSource);
```

Comment:
Any idea why that doesn't work on Arcon? You pointing to an old Kompact version where it wasn't available, yet?

Reply:
Was it added recently? Could be that we need to do a cargo update (as we are currently using git master as dependency)

Reply:
I mean...a few weeks ago...I don't know when you did cargo update last.

Anyway, if you do it at some point, try of that works, maybe and if yes delete the other variant. I can't remember if I once again forgot about supporting components with those pesky generic type parameters :D